monitors/test/index.js

```diff
@@ -0,0 +1,121 @@
+const TokenFactory = artifacts.require("TokenFactory");
+const Token = artifacts.require("ExpandedERC20");
+const Timer = artifacts.require("Timer");
+const UniswapMock = artifacts.require("UniswapMock");
+
+contract("index.js", function(accounts) {
+  const contractCreator = accounts[0];
+
+  let collateralToken;
+  let syntheticToken;
+  let emp;
+  let uniswap;
+
+  let defaultUniswapPricefeedConfig;
+  let defaultMedianizerPricefeedConfig;
+  let defaultBotMonitorConfig;
+  let defaultWalletMonitorConfig;
+  let defaultContractMonitorConfig;
+  let defaultSyntheticPegMonitorConfig;
+
+  before(async function() {
+    collateralToken = await Token.new("UMA", "UMA", 18, { from: contractCreator });
+
+    // Create identifier whitelist and register the price tracking ticker with it.
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    await identifierWhitelist.addSupportedIdentifier(utf8ToHex("UMATEST"));
+  });
+
+  beforeEach(async function() {
+    const constructorParams = {
+      expirationTimestamp: "12345678900",
+      withdrawalLiveness: "1000",
+      collateralAddress: collateralToken.address,
+      finderAddress: Finder.address,
+      tokenFactoryAddress: TokenFactory.address,
+      priceFeedIdentifier: utf8ToHex("UMATEST"),
+      syntheticName: "Test UMA Token",
+      syntheticSymbol: "UMATEST",
+      liquidationLiveness: "1000",
+      collateralRequirement: { rawValue: toWei("1.2") },
+      disputeBondPct: { rawValue: toWei("0.1") },
+      sponsorDisputeRewardPct: { rawValue: toWei("0.1") },
+      disputerDisputeRewardPct: { rawValue: toWei("0.1") },
+      minSponsorTokens: { rawValue: toWei("1") },
+      timerAddress: Timer.address
+    };
+
+    // Deploy a new expiring multi party
+    emp = await ExpiringMultiParty.new(constructorParams);
+
+    syntheticToken = await Token.at(await emp.tokenCurrency());
+
+    uniswap = await UniswapMock.new();
+
+    defaultUniswapPricefeedConfig = {
+      type: "uniswap",
+      uniswapAddress: uniswap.address,
+      twapLength: 1,
+      lookback: 1
+    };
+
+    defaultMedianizerPricefeedConfig = {
+      type: "medianizer",
+      apiKey: "test-apikey",
+      pair: "ethbtc",
+      lookback: 1,
+      minTimeBetweenUpdates: 1,
+      medianizedFeeds: [
+        {
+          type: "cryptowatch",
+          exchange: "binance"
+        }
+      ]
+    };
+
+    defaultBotMonitorConfig = [];
+    defaultWalletMonitorConfig = [];
+    defaultContractMonitorConfig = { monitoredLiquidators: [], monitoredDisputers: [] };
+    defaultSyntheticPegMonitorConfig = {
+      deviationAlertThreshold: 0.5,
+      volatilityWindow: 600,
+      volatilityAlertThreshold: 0.1
+    };
+
+    // Set two uniswap prices to give it a little history.
+    await uniswap.setPrice(toWei("1"), toWei("1"));
+    await uniswap.setPrice(toWei("1"), toWei("1"));
+  });
+
+  it("Completes one iteration without throwing an error", async function() {
```

Comment:
Perfect! thanks for adding this test. Also good that we kept it to 1 test as PR #1608 introduces some nasty restrictions around the `index.js` tests.